Yoast/wordpress-seo
# Fix options, add tests

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug introduced in 12.8 where caching of our options would lead to the meta box not showing on new custom post types and taxonomies.

## Relevant technical choices:

* Replaced several calls to `self::` withint the `WPSEO_Options` class with `static::` to prevent late static binding issues in the newly introduced tests.

## Test instructions

This PR can be tested by following these steps:

* Register a new post type, see it gets Yoast SEO Metabox. See #14181 

## Quality assurance


Fixes #14181
